blockdag/reachability.go

```diff
@@ -547,9 +811,276 @@ func (dag *BlockDAG) updateReachability(node *blockNode, selectedParentAnticone
+		return nil, nil, errors.Errorf("chosenChild not a child of rtn")
+	}
+	return rtn.children[:chosenIndex], rtn.children[chosenIndex+1:], nil
+}
+
+func (dag *BlockDAG) tightenReachabilityTreeIntervalsBeforeChosenReindexRootChild(
+	reindexRoot *reachabilityTreeNode, reindexRootChildNodesBeforeChosen []*reachabilityTreeNode) (
+	reindexRootChildNodesBeforeChosenSizesSum uint64, modifiedTreeNodes map[*reachabilityTreeNode]struct{}, err error) {
+
+	reindexRootChildNodesBeforeChosenSizes, reindexRootChildNodesBeforeChosenSubtreeSizeMaps, reindexRootChildNodesBeforeChosenSizesSum :=
+		calcReachabilityTreeNodeSizes(reindexRootChildNodesBeforeChosen)
+
+	reindexRootStart := reindexRoot.interval.start
+	targetRangeBeforeReindexRootStart := reindexRootStart + reachabilityReindexSlack
+	targetRangeBeforeReindexRootEnd := targetRangeBeforeReindexRootStart + reindexRootChildNodesBeforeChosenSizesSum - 1
+	intervalBeforeReindexRootStart := newReachabilityInterval(targetRangeBeforeReindexRootStart, targetRangeBeforeReindexRootEnd)
+
+	modifiedTreeNodes, err = dag.propagateChildIntervals(intervalBeforeReindexRootStart, reindexRootChildNodesBeforeChosen,
+		reindexRootChildNodesBeforeChosenSizes, reindexRootChildNodesBeforeChosenSubtreeSizeMaps)
+	if err != nil {
+		return 0, nil, err
+	}
+	return reindexRootChildNodesBeforeChosenSizesSum, modifiedTreeNodes, nil
+}
+
+func (dag *BlockDAG) tightenReachabilityTreeIntervalsAfterChosenReindexRootChild(
+	reindexRoot *reachabilityTreeNode, reindexRootChildNodesAfterChosen []*reachabilityTreeNode) (
+	reindexRootChildNodesAfterChosenSizesSum uint64, modifiedTreeNodes map[*reachabilityTreeNode]struct{}, err error) {
+
+	reindexRootChildNodesAfterChosenSizes, reindexRootChildNodesAfterChosenSubtreeSizeMaps, reindexRootChildNodesAfterChosenSizesSum :=
+		calcReachabilityTreeNodeSizes(reindexRootChildNodesAfterChosen)
+
+	reindexRootEnd := reindexRoot.interval.end
+	targetRangeAfterReindexRootEnd := reindexRootEnd - reachabilityReindexSlack
+	targetRangeAfterReindexRootStart := targetRangeAfterReindexRootEnd - reindexRootChildNodesAfterChosenSizesSum
+	intervalAfterReindexRootEnd := newReachabilityInterval(targetRangeAfterReindexRootStart, targetRangeAfterReindexRootEnd-1)
+
+	modifiedTreeNodes, err = dag.propagateChildIntervals(intervalAfterReindexRootEnd, reindexRootChildNodesAfterChosen,
+		reindexRootChildNodesAfterChosenSizes, reindexRootChildNodesAfterChosenSubtreeSizeMaps)
+	if err != nil {
+		return 0, nil, err
+	}
+	return reindexRootChildNodesAfterChosenSizesSum, modifiedTreeNodes, nil
+}
+
+func (dag *BlockDAG) expandReachabilityTreeIntervalInChosenReindexRootChild(reindexRoot *reachabilityTreeNode,
+	chosenReindexRootChild *reachabilityTreeNode, reindexRootChildNodesBeforeChosenSizesSum uint64,
+	reindexRootChildNodesAfterChosenSizesSum uint64) (map[*reachabilityTreeNode]struct{}, error) {
+
+	modifiedTreeNodes := make(map[*reachabilityTreeNode]struct{})
+
+	reindexRootStart := reindexRoot.interval.start
```

Comment:
Here, and in other places, I'd rather inline `reindexRoot.interval.start` instead of assign it to variable.
This way it's clearer where the var comes from.